cmd/dagger/main.go

```diff
@@ -140,13 +155,88 @@ var rootCmd = &cobra.Command{
 	},
 }
 
+var Frontend = idtui.New()
+
+func parseGlobalFlags() []string {
+	filtered := []string{}
+	for _, arg := range os.Args[1:] {
+		switch arg {
+		case "-v":
+			verbosity = 1
+		case "-vv":
+			verbosity = 2
+		case "-vvv":
+			verbosity = 3
+		case "--debug":
+			debug = true
+		case "--progress=plain":
```

Comment:
Yeah, this part is pretty annoying.

What we want is that for `dagger call -vvv -m github.com/foo/bar fizz --buzz 123` we can see the `-vvv` part and apply that change to the frontend.

We can't use the regular flag parsing stack because that all happens in `Execute` which will start initializing modules and stuff before we can even initialize the frontend and tell it to display with a verbosity level of 3.

And `pflags` doesn't expose a way to ignore extra flags; if you try to parse the above with a smaller set of global flags it'll error because it doesn't know the `--buzz` flag.

Also, if we actually register a flag called `"verbose"` that will conflict with a module-provided `verbose` flag. This is technically true for any module-provided flag whose name collides with a global flag, I just happened to run into this while testing certain modules, and we didn't have a `--verbose` flag before. The only way to really avoid this is either detecting conflicts and renaming them (kinda gross), or _only_ use short flags for these 'global things', which  we can _almost_ do, except `pflag` doesn't let you configure _only_ short flags - you always have to register a long flag.

I considered calling the long flags something stupid like `--please-verbose` but at that point recognized my desperation and went back to the comfort of this much smaller, slightly more comfortable rat's nest that is this `for {}` loop. :sob:

Reply:
Ok, phew, found a less janky solution to this: https://github.com/dagger/dagger/pull/6835/commits/d95d6641b7e5e43fcefa77bc8abe0972f42abcc5